databases/ClickHouse/Portfile

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github  1.0
+PortGroup           cmake   1.1
+
+github.setup        ClickHouse ClickHouse 22.4.2.1 v -stable
+
+description         ClickHouse® is a free analytics DBMS for big data
+
+long_description \
+    ClickHouse® is an open-source column-oriented database management \
+    system that allows generating analytical data reports in real-time.
+
+notes {
+    Install the ClickHouse-server port to run ClickHouse as a system service.
+}
+
+categories          databases
+maintainers         @slarew openmaintainer
+license             Apache-2
+platforms           darwin
+
+# clickhouse overrides the deployment target so match it
+macosx_deployment_target 10.15
+
+installs_libs       no
+
+# require vanilla clang
+compiler.whitelist  macports-clang-13 macports-clang-14
+
+cmake.build_type    RelWithDebInfo
+cmake.generator     Ninja
+
+variant debug description "Enable debug binaries" {
+    cmake.build_type    Debug
+}
```

Comment:
The cmake portgroup already defines a variant called debug. I don't think I've seen a port try to redefine a variant before and am not sure exactly what will happen. In a quick test, it seems like the new variant replaces the old one, so by doing this, you're losing all the standard settings that come with the cmake portgroup's debug variant. Is that what you want? If not—if you want to add to those settings instead—use `if {[variant_isset debug]} { ... }`.